src/features/search/components/Search.tsx

```diff
@@ -0,0 +1,137 @@
+
+const Search = (): JSX.Element => {
+    const dispatch = useAppDispatch()
+    const [searchValue, setSearchValue] = useState<string>('')
+    const recentSearches = useAppSelector(selectRecentSearches)
+
+    const onSubmit = () => {
+        searchValue &&
+            dispatch(addRecentSearch(searchValue))
+        push('Homepage', { searchValue })
+    }
+
+    const onRemoveRecentSearch = (value: string) => {
+        dispatch(removeRecentSearch(value))
+    }
+
+    const handleChange = (text: string) => {
+        setSearchValue(text)
+        dispatch(sortRecentSearches(text))
+    }
+
+    return (
+        <View style={styles.screenContainer}>
+            <TopBar styleProps={styles.topBar}>
+                <View style={styles.topBarLeft}>
+                    <GoBackButton icon={<BackIcon />} withText={false} />
+                    <TextInput
+                        placeholder="Search"
+                        style={styles.input}
+                        placeholderTextColor={Colors.BLUE400_OPACITY}
+                        value={searchValue}
+                        onChangeText={handleChange}
+                        onSubmitEditing={onSubmit}
+                    />
+                </View>
+                {searchValue &&
+                    <Pressable onPress={() => { setSearchValue('') }}>
+                        <CancelIcon />
+                    </Pressable>
+                }
+            </TopBar>
+            <MainContainer>
+                <View style={styles.headerConatiner}>
+                    <AppText styleProps={styles.header}>{Strings.RECENTS}</AppText>
+                    {(recentSearches.length !== 0) && <Pressable
+                        onPress={() => { dispatch(clearRecentSearches()) }}
+                        style={styles.clearBtn}
+                    >
+                        <AppText isBold={true}>CLEAR</AppText>
+                    </Pressable>}
+                </View>
+                {(recentSearches.length === 0)
+                    ? <AppText styleProps={styles.noRecents}>{Strings.NO_RECENTS}</AppText>
+                    : <View>
+                        {recentSearches.map((item, index) => (
+                            <Pressable
+                                onPress={() => { push('Homepage', { searchValue: item }) }}
+                                style={styles.recentsItem}
+                                key={index}
+                            >
+                                <AppText>{item}</AppText>
+                                <Pressable onPress={() => { onRemoveRecentSearch(item) }}>
+                                    <CancelIcon />
+                                </Pressable>
+                            </Pressable>
+                        ))}
+                    </View>
+                }
+            </MainContainer >
+        </View >
+    )
+}
```

Comment:
Split this return to smaller functions within this file the return should look like this: 
`<View>
{renderHeader()}
{renderSection1()}
{renderSection2()}
</View>`
etc.